### PR TITLE
Add support for optional match fields

### PR DESCRIPTION
**Pull request by daniele-moro**

### CODE REVIEW

File: p4runtime_sh/shell.py [hunk 1]
```diff
@@ -307,6 +307,7 @@ def _set_docstring(self):
   * For ternary match: <self>['<f>'] = '<value>&&&<mask>'
   * For LPM match: <self>['<f>'] = '<value>/<mask>'
   * For range match: <self>['<f>'] = '<value>..<mask>'
+  * For exact match: <self>['<f>'] = '<value>'
```
Comment (jfingerh) [inline p4runtime_sh/shell.py:310]: exact -> optional here?   exact already listed above

**Comment by daniele-moro** [comment]
I was going to add at least a very simple test for the optional match. Is the `unittest.p4info.pb.txt` being generated? or should I manually add a new table with the optional match?

**Comment by antoninbas** [comment]
@daniele-moro you may just want to update the P4Info file by hand. I believe that originally it was generated from https://github.com/p4lang/PI/blob/main/tests/testdata/unittest.p4, but it's possible that the 2 have diverged since. I know that unittest.p4info.pb.txt was updated by hand in a recent PR.
Alternatively, if you want, you can check-in unittest.p4 and make the necessary modifications to produce the "right" P4Info.

**Comment by daniele-moro** [comment]
@antoninbas Thanks for the clarification. In the end, I copied the `OptionalOne` table from the `unittest.p4info.txt` in the PI repo and used that for tests.

**Pull request merged**